### DISCUSSION TITLE
Erasing content of a given color

### DISCUSSION THREAD

**Discussion by git2212** [Bodmer/TFT_eSPI#2639] [Q&A - Sprites]
Is there a way to erase current content in a sprite (e.g. text already written) that matches a particular TFT_ color? I am aware I could clear the sprite and start all over again but being able to blank out a specific color content similar to a bitwise AND operation is what I am looking for.

Thx in advance!

**Comment by JonRobert** [comment]
I'm a new noobie with regard to TFT LCD's and Bodmer's great library.   There is a test program in the examples that tests the ability to read a pixel value from the display.  You might take a look and see if you can read the pixel bit and change if it is a target color.   I have no idea on the speed of such operation.

Good luck

**Comment by git2212** [comment]
Yes, the running time of that approach best/worst case would not be trivial... Read-compare-write sequence may be optimized quite well by the compiler... if there is no other (more direct) way, I may have to run some tests doing just that

**Comment by Bodmer** [comment]
One simple option is to write the text again using the background colour.
